### PR TITLE
docs: Shareable configs page edits and expansion

### LINKED ISSUE #16815: Change Request: Shareable Configs clean up and expansion
### ESLint version

latest

### What problem do you want to solve?

clean up/expand Shareable Configs page

### What do you think is the correct solution?

For Shareable Configs page: 

- add overview to page
- At the top of pages explaining what an extension type is, and why you might want to use it.
  - For example, in the [custom formatters documentation](https://eslint.org/docs/latest/developer-guide/working-with-custom-formatters), explain why someone might want to create a custom formatter.
- copy edit

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by bpmutter**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

light copy edits on the Shareable Configs page

#### Is there anything you'd like reviewers to focus on?

Resolves #16815

### CODE REVIEW

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

File: docs/src/extend/shareable-configs.md
Comment (nzakas) [inline docs/src/extend/shareable-configs.md:11]: Because we will be renaming the config file, let's go generic here.
```suggestion
To share your configuration file, create a **shareable config**. You can publish your shareable config on [npm](https://www.npmjs.com/) so that others can download and use it in their ESLint projects.
```

File: docs/src/extend/shareable-configs.md
Comment (nzakas) [inline docs/src/extend/shareable-configs.md:45]:
```suggestion
Once your shareable config is ready, you can [publish it to npm](https://docs.npmjs.com/getting-started/publishing-npm-packages) to share it with others. We recommend using the `eslint` and `eslintconfig` [keywords](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#keywords) in the `package.json` file so others can easily find your module.
```

**Review by nzakas — COMMENTED** [review]
Looks good, just left a couple suggestions.

File: docs/src/extend/shareable-configs.md
Comment (mdjermanovic) [inline docs/src/extend/shareable-configs.md:116]: This isn't true, a `@scope/eslint-config-myconfig` package can be used by specifying `"extends": "@scope/myconfig"`. The only case where `"eslint-config"` sequence must appear in the config specifier is when specifying a subpath of a `@scope/eslint-config` package, like in the example on line 150 (`"extends": "@scope/eslint-config/my-special-config"`), because `"@scope/my-special-config"` would be interpreted as `"@scope/eslint-config-my-special-config"`.

File: docs/src/extend/shareable-configs.md
Comment (mdjermanovic) [reply docs/src/extend/shareable-configs.md:11]: I think we should avoid "file" because shareable configs cannot be used with [`--config`](https://eslint.org/docs/latest/use/command-line-interface#-c---config) and this is rather sharing a configuration object than sharing a file.

 ```suggestion
To share your ESLint configuration, create a **shareable config**. You can publish your shareable config on [npm](https://www.npmjs.com/) so that others can download and use it in their ESLint projects.
```

Thoughts?

File: docs/src/extend/shareable-configs.md
Comment (mdjermanovic) [inline docs/src/extend/shareable-configs.md:75]: I'd keep the "Shareable configs are designed to work with the `extends` feature of `.eslintrc` files." sentence so that it isn't expected that they can be used in other ways. For example, we had a bug report that shareable configs don't work with `--config` (https://github.com/eslint/eslint/issues/15697).

File: docs/src/extend/shareable-configs.md
Comment (nzakas) [reply docs/src/extend/shareable-configs.md:11]: Makes sense to me.

File: docs/src/extend/shareable-configs.md
Comment (nzakas) [reply docs/src/extend/shareable-configs.md:116]: You're correct. I think this is a case where the old docs were also incorrect, so probably a good idea to clean this up.

File: docs/src/extend/shareable-configs.md
Comment (bpmutter) [reply docs/src/extend/shareable-configs.md:75]: i think the proposed changes are pretty clear that the shareable config should be in the ESLint config file. 

Do you think we should add some sentence like: "You cannot use shareable configs with the ESLint CLI `--config` flag"?

File: docs/src/extend/shareable-configs.md
Comment (bpmutter) [reply docs/src/extend/shareable-configs.md:116]: ok, clarifying the text and examples

**Comment by bpmutter** [comment]
@nzakas and @mdjermanovic ready for you to take another look.

File: docs/src/extend/shareable-configs.md
Comment (mdjermanovic) [reply docs/src/extend/shareable-configs.md:75]: > Do you think we should add some sentence like: "You cannot use shareable configs with the ESLint CLI `--config` flag"?

Yes, I think that would be good.

(although, technically it is possible to use a shareable config package with `--config`, e.g. `eslint --config node_modules/eslint-config-airbnb/index.js`)

Comment (nzakas) [reply docs/src/extend/shareable-configs.md:75]: I think there's still some ambiguity with the current wording. "Include it in the extends field". What is "it"? (We know it's a string, but maybe "it" should "the config name" or something more specific?)

File: docs/src/extend/shareable-configs.md
Comment (mdjermanovic) [inline docs/src/extend/shareable-configs.md:91]:
```suggestion
You cannot use shareable configs with the ESLint CLI [`--config`](../use/command-line-interface#-c---config) flag.
```

I'd leave just "you cannot" because this is not the intended use of shareable configs, and depends on internal structures of shareable config packages so we shouldn't recommend it in any way.

**Review by mdjermanovic — CHANGES_REQUESTED** [review]

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! I'm leaving this open for @nzakas to verify.

**Review by nzakas — APPROVED** [review]
LGTM. Just need to wait for the patch release before merging.

**Pull request merged**
